### PR TITLE
Faster circle and line algos

### PR DESCRIPTION
**Pull request by ganglio**
I have added Midpoint Circle Algorithm and Bresenham's algorithm to draw circles and lines faster.

I created some benchmarks and here are the results:\

BenchmarkBuffer_Line-8         	18535812	        61.44 ns/op
BenchmarkBuffer_LineFast-8     	35687457	        34.03 ns/op
BenchmarkBuffer_Circle-8       	 1000000	      1028 ns/op
BenchmarkBuffer_CircleFast-8   	20413906	        58.19 ns/op

### CODE REVIEW

File: drawings.go [hunk 1]
```diff
@@ -55,11 +55,77 @@ func (b *Buffer) Line(x1, y1, x2, y2 int, color int) {
 	}
 }
 
+// Line - draw line using the Bresenham's algorithm
+func (b *Buffer) LineFast(x1, y1, x2, y2 int, color int) {
```
Comment (msoap) [inline drawings.go:59]: let's replace the old versions with new ones

**Comment by msoap** [comment]
that's cool!

File: drawings.go (hunk 1 above)
Comment (ganglio) [reply drawings.go:59]: That was my plan. I left the old one in in case you wanted to keep them. It's your project, it will always be your last call 😀

Comment (ganglio) [reply drawings.go:59]: Now I'm looking into implementing a good flood fill algo. More pull requests will follow!

Comment (ganglio) [reply drawings.go:59]: I actually checked the code of the current flood fill implementation and it's good. I'm looking into other raster primitives that might be handy. I was thinking Bézier curves (quadratical and cubical)

**Pull request merged**